### DISCUSSION TITLE
[Question][Hw6] 첫 코드에 에러가 있습니다.

### DISCUSSION THREAD

**Discussion by HLe4s-kaist** [prosyslab-classroom/cs524-program-analysis#312] [Q&A]
Name: 강찬우

HW6를 시작하려고 했는데, 처음 clone하자마자 make를 하니, 에러가 뜹니다. src/semantics.ml에서도 에러가 나지만. 이건 사소한 문제 같습니다. 그런데 src/analysis.ml에서 
File "src/analysis.ml", line 33, characters 49-70:
33 |     let first_label = Llvmutils.find_main llm |> Llvmutils.first_label in
                                                      ^^^^^^^^^^^^^^^^^^^^^
Error: Unbound value Llvmutils.first_label
다음 에러가 뜹니다. 이것은 entry_point 함수를 쓰면 해결될거 같기는 한데
그냥 제가 해결을 해도 되는 문제일까요?

I was about to start HW6, but as soon as I cloned it and ran make, I got an error. It seems there's an error in src/semantics.ml, but it appears to be a minor issue. However, in src/analysis.ml, I encountered the following error:
File "src/analysis.ml", line 33, characters 49-70:
33 |     let first_label = Llvmutils.find_main llm |> Llvmutils.first_label in
                                                      ^^^^^^^^^^^^^^^^^^^^^
Error: Unbound value Llvmutils.first_label
It seems like using the entry_point function might resolve this, but is this an issue I can tackle on my own?

**Comment by Re-st** [comment]
세팅을 못했네요. 죄송합니다. 아래처럼 해서 llvmutils를 다시 깔아주시겠어요? 
[Llvmutils.first_label](https://prosyslab-classroom.github.io/llvmutils/llvmutils/Llvmutils/index.html#val-first_label) 가 새 함수라서 그런 것 같습니다. 

---

I'm sorry that we missed to set up. Can you do following?
```
opam remove llvmutils
opam install llvmutils
```
[Llvmutils.first_label](https://prosyslab-classroom.github.io/llvmutils/llvmutils/Llvmutils/index.html#val-first_label) is a new function, which would not be in old llvmutils.

**Comment by Re-st** [comment]
그리고 `src/semantics.ml`에서 어디가 문제인지도 보이네요. 스켈레톤 코드에 찌꺼기가 남아 있어서 그렇네요.

---
Also, you are correct in the error of `src/semantics.ml`. The skeleton was dirty with undeleted case switches.

**Comment by HLe4s-kaist** [comment]
아 좀 오래 걸리긴 하는데 말씀해주신 명령어 입력하니까 되네요. 감사합니다.

It works ! thank you
